src/math/dd/dd_pdd.cpp

```diff
@@ -51,6 +51,7 @@ namespace dd {
     void pdd_manager::reset(unsigned_vector const& level2var) {
         reset_op_cache();
         m_factor_cache.reset();
+        m_factor_cache_generation++;
```

Comment:
yes, makes sense

Reply:
would make sense to call it gc_generation. Note that m_factor_cache is reset during gc. This is the reset that could be problematic. This reset is not used during operations.